packages/test-scenarios/__tests__/scenario-test.js

```diff
@@ -90,39 +85,86 @@ module.exports = function (defaults) {
   });
 }
 
-async function embroiderInRepoAddon(project) {
-  await embroider(project);
-  await addInRepoAddon(project, 'fake-addon');
-}
+// async function embroiderInRepoAddon(project) {
+//   await embroider(project);
+//   await addInRepoAddon(project, 'fake-addon');
+// }
 
-async function addInRepoAddon(project, name, version = '0.0.0') {
+async function addInRepoAddon(project, names, version = '0.0.0') {
   project.linkDependency('ember-add-in-repo-tests', {
     baseDir: __dirname,
   });
 
   project.pkg['ember-addon'] = {
-    paths: [`lib/${name}`],
+    paths: [],
   };
 
-  merge(project.files, {
-    lib: {
-      [name]: {
-        'package.json': {
-          name,
-          version,
-          keywords: ['ember-addon'],
-        },
-        'index.js': `module.exports = {
-          name: require("./package").name,
-        };`,
-        tests: {
-          unit: getTestFiles('with-hooks-assert-includes-test.js'),
+  const lib = {};
+
+  names.forEach((name) => {
+    project.pkg['ember-addon'].paths.push(`lib/${name}`);
+    lib[name] = {
+      'package.json': `{
+        "name": "${name}",
+        "version": "${version}",
+        "keywords": ["ember-addon"]
+      }`,
+      'index.js': `module.exports = {
+        name: '${name}',
+        isDevelopingAddon() {
+          return true;
         },
+        includeTestsInHost: true,
+      };`,
+      tests: {
+        unit: getAddonTestFile('with-hooks-in-repo-addon-test.js', name),
       },
-    },
+    };
+  });
+
+  merge(project.files, {
+    'ember-cli-build.js': `'use strict';
+
+    const EmberApp = require('ember-cli/lib/broccoli/ember-app');
+    const { addInRepoTestsToHost } = require('ember-add-in-repo-tests');
+    const { getProjectConfiguration } = require('babel-plugin-ember-test-metadata/utils');
+
+    module.exports = function (defaults) {
+      let app = new EmberApp(defaults, {
+        trees: {
+          tests: addInRepoTestsToHost(
+            defaults.project,
+            addon => addon.includeTestsInHost
+          ),
+        },
+        babel: {
+          plugins: [
+            [
+              require.resolve('babel-plugin-ember-test-metadata'),
+              {
+                enabled: true,
+                projectConfiguration: getProjectConfiguration(defaults.project)
+              }
+            ]
+          ],
+        }
+      });
+
+      return app.toTree();
+    };
+    `,
+    lib,
   });
 }
 
+function getAddonTestFile(file, addonName) {
+  const result = {};
+  const tmpFile = readFileSync(join(__dirname, '__fixtures__', file), { encoding: 'utf-8' });
+
+  result[file] = tmpFile.replace('addon-name-placeholder', addonName);
+  return result;
+}
```

Comment:
You could probably just extend the other test helper that gets test files to optionally do this replacement if a param is passed in rather than having to build a whole new function to do it.